components/BottomNavigation/src/private/MDCBottomNavigationItemView.m

```diff
@@ -223,9 +222,8 @@ - (void)centerLayoutAnimated:(BOOL)animated {
       CGFloat labelCenterX =
           iconImageViewCenter.x - contentsWidth / 2 - self.contentHorizontalMargin;
       self.label.center = CGPointMake(labelCenterX, centerY);
-      self.badge.center =
-          CGPointMake(iconImageViewCenter.x - CGRectGetMidX(iconBounds),
-                      iconImageViewCenter.y - CGRectGetMidY(iconBounds));
+      self.badge.center = [self badgeCenterFrom:CGRectStandardize(self.iconImageView.frame)
```

Comment:
These 4 lines look identical. Can they moved to the end of the giant if-else construct and have a single line that does the assignment near the end of the method?